src/app/shared/search/advanced-search/advanced-search.component.ts

```diff
@@ -0,0 +1,150 @@
+import { Component, Inject, Input, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { slide } from '../../animations/slide';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { FormBuilder } from '@angular/forms';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { SearchService } from '../../../core/shared/search/search.service';
+import { RemoteData } from '../../../core/data/remote-data';
+import { SearchFilterConfig } from '../models/search-filter-config.model';
+import { SearchConfigurationService } from '../../../core/shared/search/search-configuration.service';
+import { SearchFilterService } from '../../../core/shared/search/search-filter.service';
+import { SEARCH_CONFIG_SERVICE } from '../../../my-dspace-page/my-dspace-page.component';
+import { RemoteDataBuildService } from '../../../core/cache/builders/remote-data-build.service';
+import { PaginatedSearchOptions } from '../models/paginated-search-options.model';
+import { AppConfig, APP_CONFIG } from 'src/config/app-config.interface';
+@Component({
+  selector: 'ds-advanced-search',
+  templateUrl: './advanced-search.component.html',
+  styleUrls: ['./advanced-search.component.scss'],
+  animations: [slide],
+})
+
+
+export class AdvancedSearchComponent implements OnInit {
```

Comment:
@GauravD2t 

there are a lot of missing typedocs here in this class, could you please add?

Reply:
add typedocs